nodeadm/internal/kubelet/eni_max_pods.go

```diff
@@ -1,67 +1,119 @@
 package kubelet
 
 import (
+	"bufio"
+	"bytes"
 	"context"
 	_ "embed"
-	"strconv"
-	"strings"
+	"encoding/json"
+	"fmt"
+	"math"
 
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/util"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
 	"go.uber.org/zap"
 )
 
+const (
+	defaultENIsVar = "default_enis"
+	ipsPerENIVar   = "ips_per_eni"
+	maxPodsVar     = "max_pods"
+)
+
 // default value from kubelet
 // https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration
 const defaultMaxPods = 110
 
-//go:embed eni-max-pods.txt
-var eniMaxPods string
-
-var MaxPodsPerInstanceType map[string]int
+//go:embed instance-info.jsonl
+var cachedInstanceInfoBytes []byte
 
-func init() {
-	MaxPodsPerInstanceType = make(map[string]int)
-	lines := strings.Split(eniMaxPods, "\n")
-
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		if line == "" || strings.HasPrefix(line, "#") {
+func GetInstanceInfo(ctx context.Context, awsRegion string, instanceType string) (util.InstanceInfo, error) {
+	// try to read it from the cached file first
+	cachedInfoReader := bytes.NewReader(cachedInstanceInfoBytes)
+	s := bufio.NewScanner(cachedInfoReader)
+	for s.Scan() {
+		var instanceInfo util.InstanceInfo
+		if err := json.Unmarshal(s.Bytes(), &instanceInfo); err != nil {
+			zap.L().Warn("Failed to read instance info line as json, searching in next line...", zap.Error(err))
 			continue
 		}
-
-		parts := strings.Fields(line)
-		if len(parts) != 2 {
-			continue
+		if instanceInfo.InstanceType == instanceType {
+			return instanceInfo, nil
 		}
+	}
+	zap.L().Warn("Could not find instance info locally, attempting API request...", zap.String("instanceType", instanceType), zap.String("region", awsRegion))
```

Comment:
nit: trying to use the same terminology as the [vpc cni](https://github.com/aws/amazon-vpc-cni-k8s/blob/a49107121e4c0337372dbe18d65f15d761973ebf/pkg/awsutils/awsutils.go#L1667)

```suggestion
	zap.L().Warn("Could not find instance info locally, making EC2 API call...", zap.String("instanceType", instanceType), zap.String("region", awsRegion))
```